modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/expressions/IgnitePredicateTest.java

```diff
@@ -0,0 +1,1172 @@
+
+            assertThat(predicate.test(context, row(early)), is(true));
+            assertThat(predicate.test(context, row(noon)), is(true));
+            assertThat(predicate.test(context, row(late)), is(true));
+            assertThat(predicate.test(context, row(null)), is(false));
+        }
+    }
+
+    @Test
+    @TestFor(ColumnType.DATETIME)
+    void datetimeTest() throws Exception {
+        EvaluationContext<Object[]> context = context();
+        StructNativeType inputType = singleColumnType(NativeTypes.datetime(6));
+
+        LocalDateTime past = LocalDateTime.of(2020, 1, 1, 10, 0, 0);
+        LocalDateTime present = LocalDateTime.of(2023, 6, 15, 12, 0, 0);
+        LocalDateTime future = LocalDateTime.of(2025, 12, 31, 14, 0, 0);
+
+        {
+            IgnitePredicate predicate = factory.predicate("val > TIMESTAMP '2023-06-15 12:00:00'", inputType);
+
+            assertThat(predicate.test(context, row(past)), is(false));
+            assertThat(predicate.test(context, row(present)), is(false));
+            assertThat(predicate.test(context, row(future)), is(true));
+            assertThat(predicate.test(context, row(null)), is(false));
+        }
+
+        {
+            IgnitePredicate predicate = factory.predicate("val <> TIMESTAMP '2023-06-15 12:00:00'", inputType);
+
+            assertThat(predicate.test(context, row(past)), is(true));
+            assertThat(predicate.test(context, row(present)), is(false));
+            assertThat(predicate.test(context, row(future)), is(true));
+            assertThat(predicate.test(context, row(null)), is(false));
+        }
+
+        {
+            IgnitePredicate predicate = factory.predicate("val <= TIMESTAMP '2023-06-15 12:00:00'", inputType);
+
+            assertThat(predicate.test(context, row(past)), is(true));
+            assertThat(predicate.test(context, row(present)), is(true));
+            assertThat(predicate.test(context, row(future)), is(false));
+            assertThat(predicate.test(context, row(null)), is(false));
+        }
+
+        {
+            IgnitePredicate predicate = factory.predicate("val IS NULL", inputType);
+
+            assertThat(predicate.test(context, row(past)), is(false));
+            assertThat(predicate.test(context, row(present)), is(false));
+            assertThat(predicate.test(context, row(future)), is(false));
+            assertThat(predicate.test(context, row(null)), is(true));
+        }
+
+        {
+            IgnitePredicate predicate = factory.predicate("val IS NOT NULL", inputType);
+
+            assertThat(predicate.test(context, row(past)), is(true));
+            assertThat(predicate.test(context, row(present)), is(true));
+            assertThat(predicate.test(context, row(future)), is(true));
+            assertThat(predicate.test(context, row(null)), is(false));
+        }
+    }
+
+    @Test
+    @TestFor(ColumnType.TIMESTAMP)
```

Comment:
Should we check "timetsamp with local timezone" literals ?

Reply:
a small clarification,
`TIMESTAMP WITH LOCAL TIME ZONE '2023-06-15 00:00:00'` constructs literal `TimestampWithTimeZoneString` but yes it cannot be created without timezone